crypto/s2n_openssl.h

```diff
@@ -40,3 +40,9 @@
 #else
 #define s2n_evp_ctx_init(ctx) EVP_CIPHER_CTX_init(ctx)
 #endif
+
+#if !defined(OPENSSL_IS_BORINGSSL) && !defined(OPENSSL_FIPS) && !defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
This is a fragile pattern. The lack of a way of saying "this is plain OpenSSL" makes this dangerous. What happens the next time someone adds another "not really OpenSSL" option that this isn't checking for? At least this check is more centralized compared to before though.